### PR TITLE
CANS-919 Assessment form completion for age 6 to 21 regression test

### PR DESCRIPTION
**Pull request by zliuwd**
## JIRA
https://osi-cwds.atlassian.net/browse/CANS-919
## Description

- regression test for form completion 6 to 21
- refactor form completion 0 to 5 with site_prism

## Tests

- [ ] I used TDD to develop the feature
- [ ] I have included unit tests
- [x] I have included new acceptance tests or modified existing ones
- [ ] I have included other tests

## Types of changes

- [x] New feature (non-breaking change which adds functionality)
- [x] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] I ran all my tests locally which were green.
- [x] My code adds no new issues to Code Climate
- [x] I ran all the linters for the project.
- [x] My code is in a stable state ready to be deployable, but not necessarily complete.
- [x] I promise on my honor as a CWDS developer to ensure I don't break the build (or bring donuts if I leave things broken), to check linters, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

**Comment by zliuwd** [comment]
retest this please

File: spec/regression/case_worker_scenarios_spec.rb [hunk 1]
```diff
@@ -146,45 +170,83 @@ def item_and_domain_level_comment_test
...
+    end
+    within @form.sub48a_title.sibling('h2.item-confidential-checkbox') do
+      sub48a_checkbox = find('input[type = "checkbox"]', visible: false)
+      expect(sub48a_checkbox.checked?).to be(true)
+      expect(sub48a_checkbox.disabled?).to be(true)
+    end
+    @form.header.authorization_label_yes.click
+  end
+
+  def fill_out_assessment_form_and_check_domian_total
```
Comment (wmitchell-cambria) [inline spec/regression/case_worker_scenarios_spec.rb]: `s/domian/domain/`

File: spec/page_objects/global.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+class Global < SitePrism::Section
+  element :save_button, 'div.header-buttons-block i.fa-save'
+  element :global_save_success_message_box, 'div.global-alert', text: 'Success! '\
+                                            'CANS assessment has been saved'
+  element :global_complete_message_box, 'div.global-alert', text: 'This assessment was completed '\
+                                        'and is available for view only.'
+  element :complete_warning_modal, 'div.modal', text: 'Reminder'
+  element :cancel_button_of_warning, 'button.warning-modal-logout'
+  element :agree_button_of_warning, 'button.warning-modal-stay-logged-in'
```
Comment (AK56) [inline spec/page_objects/global.rb]: The warning buttons described here also seem to be assessment specific even it's utilizing the 'warning-modal-layout' css, which is global for our app warning popups. This popup has buttons with different text in different parts of our app (timeout warning popup vs. complete warning vs remove caregiver warning). So, we should either rename our page object from 'Global' to something like 'AssessmentGlobal' or simply put them back into our AssessmentForm page object

Comment (zliuwd) [reply spec/page_objects/global.rb]: You right, I prefer AssessmentGlobal, will do

File: spec/regression/case_worker_scenarios_spec.rb [hunk 3]
```diff
@@ -146,45 +170,83 @@ def item_and_domain_level_comment_test
...
-    @form.remove_first_caregiver_domain_button.click
-    expect(@form.has_caregiver_domain_warning_popup?).to be(true)
-    expect(@form.caregiver_domain_warning_message.text).to eq(CAREGIVER_DOMAIN_WARNING_MESSAGE)
-    @form.has_caregiver_domain_warning_popup_buttons?
-    caregiver_domain_warning_remove_button.click
-    expect(@form.caregiver_name_fields.length).to eq(1)
+  def fill_form_header_0_to_5
+    @client_list.visit_client_profile(CLIENT_NAME)
+    expect(page).to have_content('ADD CANS')
+    create_assessment_form
```
Comment (AK56) [inline spec/regression/case_worker_scenarios_spec.rb]: Duplicate code: We can utilize our `create_new_assessment` method

File: spec/regression/case_worker_scenarios_spec.rb [hunk 4]
```diff
@@ -146,45 +170,83 @@ def item_and_domain_level_comment_test
...
-    expect(@form.caregiver_domain_warning_message.text).to eq(CAREGIVER_DOMAIN_WARNING_MESSAGE)
-    @form.has_caregiver_domain_warning_popup_buttons?
-    caregiver_domain_warning_cancel_button.click
-    expect(@form.caregiver_name_fields.length).to eq(1)
-    within(@form.has_caregiver_yes_label) do
-      expect(find('input', visible: false).checked?).to be(true)
+  def fill_form_header_6_to_21
+    @client_list.visit_client_profile(CLIENT_NAME)
+    expect(page).to have_content('ADD CANS')
+    create_assessment_form
```
Comment (AK56) [inline spec/regression/case_worker_scenarios_spec.rb]: Likewise here: use `create_new_assessment` method so our code follows DRY principle

Comment (zliuwd) [reply spec/regression/case_worker_scenarios_spec.rb]: good catch man!

**Pull request merged**